src/dali/mapped_graph.py

```diff
@@ -164,17 +180,43 @@ def clone_with_optimization(self, optimization: Dict[str, Dict[str, float]]) ->
                 # Delete neighbor if negative weight
                 if neighbor_weight >= 0.0:
                     cloned_mapped_graph.add_neighbor(original_vertex.name, neighbor.name, neighbor_weight, optimized)
+                elif neighbor.name in self.__fiat_assets:
+                    LOGGER.debug("Adding fiat neighbor while cloning %s to %s", original_vertex.name, neighbor.name)
+                    cloned_mapped_graph.add_fiat_neighbor(original_vertex.name, neighbor.name, neighbor_weight, optimized)
                 else:
                     cloned_mapped_graph.add_vertex_if_missing(original_vertex.name)
 
-            # Add new neighbors
-            for optimized_asset, neighbor_weights in optimization.items():
-                for neighbor_name in neighbor_weights.keys():
-                    if original_vertex.name == optimized_asset:
-                        cloned_mapped_graph.add_neighbor(original_vertex.name, neighbor_name, neighbor_weights[neighbor_name], True)
+        # Add new neighbors
+        for optimized_asset, neighbor_weights in optimization.items():
```

Comment:
Could you explain why this for has been removed from the for at line 166? It's a bit unclear to me.

Reply:
You mean why it is no longer in the for block starting at line 166?

The goal of this for loop is to add any optimizations that were not found in the graph. In the preceding for loop an **edge** optimization that is currently in the graph is popped off the dict in line 175:
https://github.com/eprbell/dali-rp2/blob/508dc43cbfc0bd99dfef85c522f3cecb6055df9d/src/dali/mapped_graph.py#L175

so, the only remaining **edges** in the optimization dict are ones that are not in the current graph and must be added. This loop goes through the leftover edges and applies them to nodes/points that are in the graph. 

Previously, it was indented because the graph had all the edges and so it would search for and replace the edges. Now the edges are deleted (not the nodes or points). so, the edges need to be added back in.